### PR TITLE
Add Slim 4 support

### LINKED ISSUE #31: Slim 4 compatibility
I made an attempt to make this Slim 4 compatible. Please check https://github.com/adriansuter/Slim-HttpCache/tree/patch-slim4

@l0gicgate I could make a PR against a new branch if you like.

README is not updated yet.

### PR DESCRIPTION
**Pull request by adriansuter**
Fixes #31

### CODE REVIEW

**Comment by l0gicgate** [comment]
README mentions of `Slim 3.0.0` need to be switched to `4.0.0` and the example needs to use `AppFactory` as well.

File: composer.json [hunk 1]
```diff
@@ -1,33 +1,51 @@
...
+  "scripts": {
+    "test": [
+      "@phpunit",
+      "@phpcs",
+      "@phpstan"
+    ],
+    "phpunit": "phpunit",
+    "phpcs": "phpcs",
+    "phpstan": "phpstan analyse src --memory-limit=-1"
+  }
```
Comment (l0gicgate) [inline composer.json]: let's order those alphabetically

**Review by l0gicgate — COMMENTED** [review]
Looks absolutely fantastic. Great work @adriansuter. Besides the README change and my one comment I don't see anything outstanding.

File: composer.json (hunk 1 above)
Comment (adriansuter) [reply composer.json]: Done

**Review by l0gicgate — APPROVED** [review]

**Comment by l0gicgate** [comment]
Thank you again for this @adriansuter amazing work as usual

**Pull request merged**